src/filter-summary/index.jsx

```diff
@@ -1,31 +1,74 @@
-import React from 'react';
+import React, { Fragment } from 'react';
 import { connect } from 'react-redux';
+import isEmpty from 'lodash/isEmpty';
+import omit from 'lodash/omit';
+import capitalize from 'lodash/capitalize';
+import qs from 'qs';
 
 const FilterSummary = ({
   total,
   filtered,
-  filteredLabel = `Showing ${filtered} of ${total} results`,
-  allShowingLabel = `All ${total} results`
-}) => (
-  <h2 className="filter-summary">
-    {
-      filtered !== total
-        ? filteredLabel
-        : allShowingLabel
-    }
-  </h2>
-);
+  filteredLabel = filtered === 1 ? `${filtered} result` : `${filtered} results`,
+  allShowingLabel = total === 1 ? `${total} result` : `${total} results`,
+  filters
+}) => {
+  const searchString = filters.active['*'] && filters.active['*'][0];
+  const activeFilters = omit(filters.active, '*');
+
+  const removeFilter = filterKey => {
+    return e => {
+      e.preventDefault();
+      const urlParams = qs.parse(window.location.search, { ignoreQueryPrefix: true });
+      delete urlParams['filters'][filterKey];
+      window.location.href = `//${window.location.host + window.location.pathname + qs.stringify(urlParams, { addQueryPrefix: true })}`;
```

Comment:
If we know what the end state of the query string after clicking a link should be then can we pre-calculate and just hard-code the links?

Reply:
Reason: be slightly less dependent on client-side script.